llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -448,6 +454,142 @@ def construct_blocks(self, config: MPTConfig) -> nn.ModuleList:
             ) for _ in range(config.n_layers)
         ])
 
+    def _construct_blocks_with_overrides(
+        self,
+        config: MPTConfig,
+        block_args: Dict[str, Any],
+    ) -> nn.ModuleList:
+        if config.block_overrides is None:
+            raise ValueError(
+                'config.block_overrides should not be None when calling _construct_blocks_with_overrides.',
+            )
+        modules_order_expanded = {}
+        for type in 'start', 'repeating_pattern', 'end':
+            modules_order_expanded[type] = []
+            if type in config.block_overrides:
```

Comment:
can combine these lines to `for block in config.block_overrides.get(type, [])`

Reply:
Done.